File: BabelsSales/src/babelssalesObjects/Product.java
package babelssalesObjects;

import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.sql.PreparedStatement;

public class Product {

    private final String TABLENAME = "Products";
    private final String FIELD_ID = "Id";
    private final String FIELD_NAME = "Name";
    private final String FIELD_PRICE = "Price";
    private Connection Conn;
    private int Id;
    public String Name;
    public float Price;

    public int getId() {
        return this.Id;
    }

    public Product(Connection conn) throws SQLException {
        this.Conn = conn;
        Clear();
    }

    public void Clear() {
        this.Id = -1;
        this.Name = "";
        this.Price = Float.parseFloat("0");
    }

    public boolean Load(Integer id) throws SQLException {
        String sql = "SELECT * FROM " + this.TABLENAME + " WHERE "
                + this.FIELD_ID + " = ?";
        PreparedStatement qry = this.Conn.prepareStatement(sql);
        try {
            qry.setInt(1, id);
            return SelectProduct(qry);
        } finally {
            qry.close();
        }
    }

    public boolean Load(String name) throws SQLException {
        String sql = "SELECT * FROM " + this.TABLENAME + " WHERE "
                + this.FIELD_NAME + " = ?";
        PreparedStatement qry = this.Conn.prepareStatement(sql);
        try {
            qry.setString(1, name);
            return SelectProduct(qry);
        } finally {
            qry.close();
        }
    }

    private boolean SelectProduct(PreparedStatement qry) throws SQLException {
        ResultSet results = qry.executeQuery();
        try {
            if (results.next()) {
                this.Id = results.getInt(this.FIELD_ID);
                this.Name = results.getString(this.FIELD_NAME);
                this.Price = results.getFloat(this.FIELD_PRICE);
                return true;
            } else {
                return false;
            }
        } finally {
            results.close();
        }
    }
}

File: trunk/BabelsPrinter/BabelsPrinter/Model/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using MySQLDriverCS;

namespace BabelsPrinter.Model
{
    public class Product
    {
        public static string TABLENAME = "Products";
        public static string FIELD_ID = "Id";
        public static string FIELD_NAME = "Name";
        public static string FIELD_DESCRIPTION = "Description";
        public static string FIELD_PRICE = "Price";
        public static string FIELD_IDKITCHEN = "IdKitchen";

        private MySQLConnection Conn;
        private int _Id;
        private string _Name;
        private string _Description;
        private double _Price;
        private int _IdKitchen;

        public int Id { get { return _Id; } set { _Id = value; } }
        public string Name { get { return _Name; } set { _Name = value; } }
        public string Description { get { return _Description; } set { _Description = value; } }
        public double Price { get { return _Price; } set { _Price = value; } }
        public int IdKitchen { get { return _IdKitchen; } set { _IdKitchen = value; } }

        public Product(MySQLConnection conn)
        {
            Conn = conn;
        }

        public void Load(int id)
        {
            string sql = "SELECT * FROM " + TABLENAME +
                " WHERE " + FIELD_ID + "= " + id.ToString();
            MySQLCommand comm = new MySQLCommand(sql, Conn);
            try
            {
                MySQLDataReader reader = comm.ExecuteReaderEx();
                if (reader.HasRows)
                {
                    reader.Read();
                    this.Id = reader.GetInt32(reader.GetOrdinal(FIELD_ID));
                    this.Name = reader.GetString(reader.GetOrdinal(FIELD_NAME));
                    this.Description = "";
                    if (!reader.IsDBNull(reader.GetOrdinal(FIELD_DESCRIPTION)))
                    {
                        this.Description = reader.GetString(reader.GetOrdinal(FIELD_DESCRIPTION));
                    }
                    this.Price = reader.GetDouble(reader.GetOrdinal(FIELD_PRICE));
                    this.IdKitchen = reader.GetInt32(reader.GetOrdinal(FIELD_IDKITCHEN));
                }
            }
            catch (Exception ex) { }
            finally
            {
                comm.Dispose();
            }
        }
    }
}
